src/cmd/tools/wait.go

```diff
@@ -56,9 +62,40 @@ var waitForCmd = &cobra.Command{
 	},
 }
 
+var waitForResourceCmd = &cobra.Command{
+	Use:     "wait-for-resource { KIND[.VERSION][.GROUP] } { NAME }",
+	Short:   lang.CmdToolsWaitForShort,
+	Long:    lang.CmdToolsWaitForLong,
+	Example: lang.CmdToolsWaitForExample,
+	Args:    cobra.MinimumNArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		timeout, err := time.ParseDuration(waitResourceTimeout)
+		if err != nil {
+			message.Fatalf(err, lang.CmdToolsWaitForErrTimeoutString, waitResourceTimeout)
+		}
+		ctx, cancel := context.WithTimeout(cmd.Context(), timeout)
+		defer cancel()
+		// Accept passing version even if we ignore it.
+		gvk, gk := schema.ParseKindArg(args[0])
+		if gvk != nil {
+			gk = gvk.GroupKind()
+		}
+		err = utils.ExecuteWaitResource(ctx, gk, waitResourceNamespace, args[1])
+		if err != nil {
+			message.Fatal(err, err.Error())
+		}
+	},
+}
+
 func init() {
 	toolsCmd.AddCommand(waitForCmd)
 	waitForCmd.Flags().StringVar(&waitTimeout, "timeout", "5m", lang.CmdToolsWaitForFlagTimeout)
 	waitForCmd.Flags().StringVarP(&waitNamespace, "namespace", "n", "", lang.CmdToolsWaitForFlagNamespace)
 	waitForCmd.Flags().BoolVar(&message.NoProgress, "no-progress", false, lang.RootCmdFlagNoProgress)
+
+	toolsCmd.AddCommand(waitForResourceCmd)
+	waitForResourceCmd.Flags().StringVar(&waitResourceTimeout, "timeout", "5m", lang.CmdToolsWaitForFlagTimeout)
+	waitForResourceCmd.Flags().StringVarP(&waitResourceKubeconfig, "kubeconfig", "f", "", lang.CmdToolsWaitForFlagNamespace)
```

Comment:
It might be a lot better to place this flag in the root command so that it applies to all sub commands. This would simplify future credential configuration.

Reply:
Flag should exist in the root command so lets leave that for a later PR.